### PR TITLE
Ofir/tfl 1246

### PR DESCRIPTION
**Pull request by OfirZinkerz**
Implement the purchase mechanism in infa-web-app

### CODE REVIEW

File: dist/invitation/main.css [hunk 1]
```diff
@@ -394,7 +394,7 @@ md-dialog.invite-teacher-modal-wrap {
         font-size: 12px;
         color: #757a83;
         font-family: Lato;
-        font-weight: 200; }
+        font-weight: 300; }
```
Comment (assafshp) [inline dist/invitation/main.css]: 200

File: src/components/purchase/services/purchase.service.js [hunk 2]
```diff
@@ -101,41 +130,34 @@
...
-                       if (purchaseData) {
-                           self.removePendingPurchase();
-                       }
-                   });
-               }
+                studentStorageProm.then(function (studentStorage) {
+                    self.hasProVersion().then(function (hasPro) {
+                        studentStorage.cleanPathCache(purchasePath);
+
+                        var removeListener = $rootScope.$on('$stateChangeSuccess', function () {
```
Comment (assafshp) [inline src/components/purchase/services/purchase.service.js]: why do you need it?

File: src/components/purchase/services/purchase.service.js [hunk 3]
```diff
@@ -2,62 +2,91 @@
...
 
             self.checkPendingStatus = function () {
-                var isPending;
-                return $q.when(studentStorageProm).then(function (StorageSrv) {
-                    var pendingPurchasesPath = 'pendingPurchases/' + StorageSrv.variables.uid;
-
-                    return StorageSrv.get(pendingPurchasesPath).then(function (pendingObj) {
-                        isPending = (angular.equals(pendingObj, {})) ? false : true;
+                return studentStorageProm.then(function (studentStorage) {
+                    return studentStorage.get(pendingPurchasesPath).then(function (pendingObj) {
```
Comment (assafshp) [inline src/components/purchase/services/purchase.service.js]: are you sure return studentStorage.get returns the pendingPurchasesPath?

File: dist/main.js [hunk 4]
```diff
@@ -7517,62 +7500,91 @@ angular.module('znk.infra-web-app.onBoarding').run(['$templateCache', function($
...
+                return path.replace('$$uid', '' + authData.uid);
+            }
+            function getPendingPath() {
+                if (!authData) {
+                    $log.error('Invalid user');
+                    return;
+                }
+                var path = 'pendingPurchases/' + StorageSrv.variables.uid;
+                return path.replace('$$uid', '' + authData.uid);
+            }
```
Comment (assafshp) [inline dist/main.js]: I think you should merge these two functions to one func with path param....they are 90% similar getPurchasePath() and getPendingPath()

File: dist/purchase/purchase.js [hunk 5]
```diff
@@ -209,58 +192,91 @@
...
                 });
             };
 
-            self.hasProVersion = function () {
-                var hasProVersion = !!purchaseData;
-                return $q.when(hasProVersion);
+            self.getUserData = function () {
+                return $q.when(studentStorageProm).then(function (studentStorage) {
+                    var userPath = 'users/' + authData.uid;
+                    return studentStorage.getAndBindToServer(userPath);
```
Comment (assafshp) [inline dist/purchase/purchase.js]: check this

File: src/components/purchase/services/purchase.service.js [hunk 6]
```diff
@@ -101,41 +130,34 @@
                 if (pendingPurchaseDefer) {
                     pendingPurchaseDefer.resolve();
                 }
-                $q.when(studentStorageProm).then(function (StorageSrv) {
-                    var pendingPurchasesPath = 'pendingPurchases/' + StorageSrv.variables.uid;
-                    return StorageSrv.set(pendingPurchasesPath, null);
+                studentStorageProm.then(function (studentStorage) {
+                    return studentStorage.set(pendingPurchasesPath, null);
```
Comment (OfirZinkerz) [inline src/components/purchase/services/purchase.service.js]: i declared it before in the top of the file

File: src/components/purchase/services/purchase.service.js (hunk 2 above)
Comment (OfirZinkerz) [inline src/components/purchase/services/purchase.service.js]: I passed it from ran's code and I never test what;s happens if I remove ot

File: src/components/purchase/services/purchase.service.js (hunk 3 above)
Comment (OfirZinkerz) [inline src/components/purchase/services/purchase.service.js]: yep

File: dist/purchase/purchase.js (hunk 5 above)
Comment (OfirZinkerz) [inline dist/purchase/purchase.js]: I removed this fn

**Pull request merged**